apps/whale-api/src/module.api/poolpair.controller.e2e.ts

```diff
@@ -527,11 +535,94 @@ describe('get best path', () => {
-          tokenB: { id: '11', name: 'K', symbol: 'K', displaySymbol: 'dK' }
+          tokenB: { id: '11', name: 'K', symbol: 'K', displaySymbol: 'dK' },
+          commissionFeeInPct: '0.25000000'
         }
       ],
       estimatedReturn: '7.00000000',
-      estimatedReturnLessDexFees: '7.00000000'
+      estimatedReturnLessDexFees: '5.25000000'
+    })
+  })
+
+  it('should deduct commission fee - 1 leg', async () => {
+    const response = await controller.getBestPath('10', '12')
+    expect(response).toStrictEqual({
+      fromToken: {
+        id: '10',
+        name: 'J',
+        symbol: 'J',
+        displaySymbol: 'dJ'
+      },
+      toToken: {
+        id: '12',
+        name: 'L',
+        symbol: 'L',
+        displaySymbol: 'dL'
+      },
+      bestPath: [
+        {
+          symbol: 'J-L',
+          poolPairId: '24',
+          priceRatio: { ab: '0.50000000', ba: '2.00000000' },
+          tokenA: { id: '10', name: 'J', symbol: 'J', displaySymbol: 'dJ' },
+          tokenB: { id: '12', name: 'L', symbol: 'L', displaySymbol: 'dL' },
+          commissionFeeInPct: '0.10000000'
+        }
+      ],
+      estimatedReturn: '2.00000000',
+      estimatedReturnLessDexFees: '1.80000000'
+    })
+  })
+
+  it('should deduct commission and dex fees - 2 legs', async () => {
+    const response = await controller.getBestPath('10', '13')
+    expect(response).toStrictEqual({
+      fromToken: {
+        id: '10',
+        name: 'J',
+        symbol: 'J',
+        displaySymbol: 'dJ'
+      },
+      toToken: {
+        id: '13',
+        name: 'M',
+        symbol: 'M',
+        displaySymbol: 'dM'
+      },
+      bestPath: [
+        {
+          symbol: 'J-L',
+          poolPairId: '24',
+          priceRatio: { ab: '0.50000000', ba: '2.00000000' },
+          tokenA: { id: '10', name: 'J', symbol: 'J', displaySymbol: 'dJ' },
+          tokenB: { id: '12', name: 'L', symbol: 'L', displaySymbol: 'dL' },
+          commissionFeeInPct: '0.10000000'
+        },
+        {
+          symbol: 'L-M',
+          poolPairId: '26',
+          priceRatio: { ab: '0.12500000', ba: '8.00000000' },
+          tokenA: { id: '12', name: 'L', symbol: 'L', displaySymbol: 'dL' },
+          tokenB: { id: '13', name: 'M', symbol: 'M', displaySymbol: 'dM' },
+          commissionFeeInPct: '0.50000000',
+          estimatedDexFeesInPct: {
+            ab: '0.09000000',
+            ba: '0.07000000'
+          }
+        }
+      ],
+      estimatedReturn: '16.00000000',
+      /*
+        Swap through first leg -- J -> L (only commission fees)
+          -> Deduct commission fee
+          -> Convert fromToken -> toToken by price ratio
+
+        Swap through second leg -- L -> M (with commission & DEX fees)
+          -> Deduct commission fee and dex fees
+          -> Convert fromToken -> toToken by price ratio
+          -> Deduct dex fees toToken
+      */
+      estimatedReturnLessDexFees: '6.09336000'
```

Comment:
```suggestion
      /*
        Swap through first leg -- J -> L (No DEX fees)
          Deduct commission fee: 1 * 0.1
            = 1 - (1 * 0.1) 
          Convert fromToken -> toToken by price ratio
            = 0.9 * 2

        Swap through second leg -- L -> M  (With DEX fees)
          Deduct commission fee: 1.8 * 0.5
            = 1.8 - 0.9
          Deduct dex fees fromToken: estLessDexFees * 0.07
            = 0.9 - 0.063
          Convert fromToken -> toToken by price ratio
            = 0.837 * 8
          Deduct dex fees toToken: estLessDexFees * 0.09
            = 6.696 - 0.60264
          
        Estimated return less commission and dex fees
          = 6.09366
      */
      estimatedReturnLessDexFees: '6.09336000'
```
Or do you prefer with actual values? 😁  @eli-lim

Reply:
Yes @pierregee but final value should be `6.09336` instead of `6.09366`

Reply:
oops okies typoo